services/sandbox/controller_service.go

```diff
@@ -72,10 +84,24 @@ func (s *controllerService) Register(server *grpc.Server) error {
 	return nil
 }
 
+func (s *controllerService) getController(name string) (sandbox.Controller, error) {
+	if len(name) == 0 {
+		name = "local"
```

Comment:
I am thinking that we should block the empty sandboxer name here. The `local` is too fragile here.
It's hard to track what is `local` here. In your patches, the `local` has been changed to `shim`, if I understand correctly. Maybe we should introduce builtin or reserved names for `local(or shim)` and `podsandboxer(or cripodsandboxer)`.

Reply:
changed to return error if the controller name is empty.